genesyscloud/telephony_providers_edges_phone/resource_genesyscloud_telephony_providers_edges_phone.go

```diff
@@ -0,0 +1,273 @@
+package telephony_providers_edges_phone
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v115/platformclientv2"
+)
+
+func getAllPhones(ctx context.Context, sdkConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	pp := getPhoneProxy(sdkConfig)
+
+	phones, err := pp.getAllPhones(ctx)
+	if err != nil {
+		return nil, diag.Errorf("Failed to get page of phones: %v", err)
+	}
+
+	for _, phone := range *phones {
+		resources[*phone.Id] = &resourceExporter.ResourceMeta{Name: *phone.Name}
+	}
+
+	return resources, nil
+}
+
+func createPhone(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	pp := getPhoneProxy(sdkConfig)
+
+	name := d.Get("name").(string)
```

Comment:
You can set properties directly onto the Phone object by combining them with platformclientv2.String(). This way we can reduce repetitive code. On the phone object try this
```
Name: platformclientv2.String(d.Get("name").(string))
```
There is similar functions for other types like boolean, int, and float

Reply:
Simplified code to use platformclientv2 methods